test/models/test_mom6.py

```diff
@@ -0,0 +1,144 @@
+import copy
+import os
+import shutil
+
+import pytest
+import f90nml
+
+import payu
+
+from test.common import cd
+from test.common import tmpdir, ctrldir, labdir, expt_workdir
+from test.common import config as config_orig
+from test.common import write_config
+from test.common import make_random_file, make_inputs
+
+verbose = True
+
+# Global config
+config = copy.deepcopy(config_orig)
+config["model"] = "mom6"
+
+
+def setup_module(module):
+    """
+    Put any test-wide setup code in here, e.g. creating test files
+    """
+    if verbose:
+        print("setup_module      module:%s" % module.__name__)
+
+    # Should be taken care of by teardown, in case remnants lying around
+    try:
+        shutil.rmtree(tmpdir)
+    except FileNotFoundError:
+        pass
+
+    try:
+        tmpdir.mkdir()
+        labdir.mkdir()
+        ctrldir.mkdir()
+        expt_workdir.mkdir(parents=True)
+        make_inputs()
+    except Exception as e:
+        print(e)
+
+    write_config(config)
+
+
+def teardown_module(module):
+    """
+    Put any test-wide teardown code in here, e.g. removing test outputs
+    """
+    if verbose:
+        print("teardown_module   module:%s" % module.__name__)
+
+    try:
+        shutil.rmtree(tmpdir)
+        print('removing tmp')
+    except Exception as e:
+        print(e)
+
+
+@pytest.fixture(autouse=True)
+def teardown():
+    # Run test
+    yield
+
+    # Remove any files in expt work directory
+    for file in os.listdir(expt_workdir):
+        try:
+            os.remove(os.path.join(expt_workdir, file))
+        except Exception as e:
+            print(e)
+
+
+def make_config_files(parameter_files):
+    """Make config files in control directory"""
+    for file in parameter_files:
+        filename = os.path.join(ctrldir, file)
+        make_random_file(filename, 8)
+
+
+def make_input_nml_file(mom_parameter_files, sis_parameter_files=None):
+    """Create an input.nml in expt work directory"""
+    input_nml = {
+        'MOM_input_nml': {
+            'parameter_filename': mom_parameter_files,
+        }
+    }
+    if sis_parameter_files:
+        input_nml['SIS_input_nml'] = {
+            'parameter_filename': sis_parameter_files,
+        }
+    input_nml_fp = os.path.join(expt_workdir, 'input.nml')
+    f90nml.write(input_nml, input_nml_fp)
+
+
+@pytest.mark.parametrize(
+    "mom_parameter_files, sis_parameter_files, expected_files_added",
+    [
+        (
+            ['MOM_input'],
+            ['SIS_input', 'SIS_layout'],
+            []
+        ),
+        (
+            ['MOM_input', 'MOM_layout'],
+            ['SIS_input', 'New_SIS_file'],
+            ['New_SIS_file']
+        ),
+        (
+            ['New_MOM_file', 'MOM_input'],
+            None,
+            ['New_MOM_file']
+        )
+    ])
```

Comment:
Maybe one of these should test the case where the `mom_parameter_files` or `sis_parameter_files` is a bare string, rather than a list (wrt lines 96-97 in mom6.py)?

Reply:
I think the `f90nml.write()` method, writes lists of one elements as a string. But I can make it more clear in the code, and explicitly use test inputs of strings rather than lists with 1 element.

Reply:
It's probably not too big an issue, just a suggestion to hit that extra code path. In fact, given that namelists are typed and the `parameter_filename` field is an array, it's probably just a quirk of f90nml and doesn't explicitly need to be tested...